dask/dataframe/groupby.py

```diff
@@ -231,12 +230,28 @@ def chunk(df, index):
                 grouped = pd.concat([df, index], axis=1).drop_duplicates()
             return grouped
 
+        key = self.key
+        is_series = isinstance(self.df, Series)
+
         def agg(df):
-            if isinstance(self.df, Series):
+            if is_series:
                 return df.groupby(df.columns[1])[df.columns[0]].nunique()
             else:
-                return df.groupby(level=0)[self.key].nunique()
+                return df.groupby(level=0)[key].nunique()
 
-        return aca([self.df, self.index],
+        return aca([self.df, self.index, self.key],
                    chunk=chunk, aggregate=agg, columns=self.key,
                    token='series-groupby-nunique')
+
+
+def _sum(x):
+    return x.sum()
```

Comment:
Optional nitpick - these could be replaced with `methodcaller` objects. Results in smaller serialized form, and faster serialization time with cloudpickle (although both are super negligible). Feel free to ignore.

```python
In [6]: import cloudpickle

In [7]: from operator import methodcaller

In [8]: _sum = methodcaller('sum')

In [9]: def _sum2(x):
   ...:     return x.sum()
   ...:

In [10]: cloudpickle.dumps(_sum)
Out[10]: '\x80\x02coperator\nmethodcaller\nq\x00)\x81q\x01.'

In [11]: cloudpickle.dumps(_sum2)
Out[11]: '\x80\x02ccloudpickle.cloudpickle\n_fill_function\nq\x00(ccloudpickle.cloudpickle\n_make_skel_func\nq\x01ccloudpickle.cloudpickle\n_builtin_type\nq\x02U\x08CodeTypeq\x03\x85q\x04Rq\x05(K\x01K\x01K\x01KCU\n|\x00\x00j\x00\x00\x83\x00\x00Sq\x06N\x85q\x07U\x03sumq\x08\x85q\tU\x01xq\n\x85q\x0bU\x1e<ipython-input-9-2e7c5b0ec5c4>q\x0cU\x05_sum2q\rK\x01U\x02\x00\x01q\x0e))tq\x0fRq\x10]q\x11}q\x12\x87q\x13Rq\x14}q\x15N}q\x16tR.'

In [12]: %timeit cloudpickle.dumps(_sum)
The slowest run took 4.28 times longer than the fastest. This could mean that an intermediate result is being cached
10000 loops, best of 3: 21.5 µs per loop

In [13]: %timeit cloudpickle.dumps(_sum2)
10000 loops, best of 3: 108 µs per loop
```